### ISSUE TITLE
Can't mirror tensorflow properly for 3.11 when 3.12 is also configured

### ISSUE THREAD

**Issue by alex-orange** [ido50/morgan#21]
If I try to mirror `tensorflow` with python environments for both 3.11 and 3.12 then it won't grab `tensorflow-io-gcs-filesystem` which is required for python <3.12. From what I can gather the trouble is that there's no version of `tensorflow-io-gcs-filesystem` for 3.12 and so morgan gives up and decides not to get it. Not exactly sure why that's not fatal, but more to the point morgan should be handling the different environments somewhat more independently. Environment a needs package x, then get a version of package x that fulfills the requirement for environment a. Don't care if it doesn't work for environment b, you still need it for environment a. If you want to get fancy it ends up being a packing problem. I would suggest just doing the trivial "find a version that works for the environment I need it in" and don't worry about trying to reduce duplication.

**Comment by grische** [comment]
I think it might be interesting to have a flag to switch between "package needs to work in all environments" and "package needs to work in either environment":

https://github.com/ido50/morgan/blob/2c5ba7c45b80adb9ac57cdf93a0eb54469f242ad/morgan/__init__.py#L286-L290

**Comment by grische** [comment]
I think this means that #21 can be closed in favour of #46

**Comment by alex-orange-UofU** [comment]
I'm not sure why. It's a duplicate of this one. I would just mark it as such and ask that this be fixed.

**Comment by alex-orange-UofU** [comment]
> I think it might be interesting to have a flag to switch between "package needs to work in all environments" and "package needs to work in either environment":
> 
> [morgan/morgan/__init__.py](https://github.com/ido50/morgan/blob/2c5ba7c45b80adb9ac57cdf93a0eb54469f242ad/morgan/__init__.py#L286-L290)
> 
> Lines 286 to 290 in [2c5ba7c](/ido50/morgan/commit/2c5ba7c45b80adb9ac57cdf93a0eb54469f242ad)
>  for supported_python in self._supported_pyversions: 
>      if not spec_set.contains(supported_python): 
>          # file does not support the Python version of one of our 
>          # environments, reject it 
>          return False

I disagree. I see no value in trying to force package works in all environments. At best this maybe saves a tidbit of space, but it likely comes at the cost of having the most up to date package on newer versions of python. I would also say that if it fails to find one package that matches all versions, it then needs to fails (i.e. with an actual message). This is where the lack of value comes in for me, why have a system where failure is an option (as a consequence of choices made, not just lack of any version that could work) vs just getting separate versions for each version.
